TACC/Core-Portal
# Fp 470/infinite scroll table css

## Overview: ##

Migrate `InfiniteScrollTable` to use a CSS Module, and make related changes.

## PR Status: ##


## Related Jira tickets: ##

* [FP-470](https://jira.tacc.utexas.edu/browse/FP-470)

## Summary of Changes: ##

1. edit `.js` to use `styleName`
1. use `o-fixed-table-header` directly as global class
1. add "micro-component" classes for table status elements
1. rename `.scss` to `module.scss`
1. edit `module.scss` to use more classes and less tags
1. allow `.scss` in `src/styles`
1. prepare a `table.css` for the future _(after `react-table-6` is dead)_
1. extract `o-fixed-table-header` to its own file
1. add mixin for sharing ellipsis styles

## Testing Steps: ##
1. Load dashboard.
1. Inspect Jobs table.
1. Ensure loading spinner has not changed compared to source branch.¹
1. Ensure no data message has not changed compared to source branch.¹
1. Ensure data rows have not changed compared to source branch.¹
1. Ensure loading spinner row, no data message, and data rows are the same height (excluding border).¹
1. Ensure last row has no bottom border.¹
1. Ensure unwrapped cell text (text without a `<span>`/`<a>`) will still have long text truncated.²

¹ You may test different rows by manipulating app state for user, or by changing code in dev by setting `tableData = []` and `isLoading = true` as needed.
² You may test plain unwrapped text table cell by removing the `a` and `span` from the Job Name cell content on the fly in the dynamic markup ([example](https://user-images.githubusercontent.com/62723358/87969214-509bb980-ca87-11ea-8b34-e6b4237e3441.png)).


## UI Photos:

* FP-470 Has Data (Tickets)
    <img width="999" alt="FP-470 Has Data" src="https://user-images.githubusercontent.com/62723358/84724564-20c03a00-af4e-11ea-9e35-b6345d6103e8.png">
* FP-470 No Data (Tickets)
    <img width="999" alt="FP-470 No Data" src="https://user-images.githubusercontent.com/62723358/84724569-2289fd80-af4e-11ea-861f-399683dfba57.png">
* FP-470 Is Loading (Tickets)
    <img width="999" alt="FP-470 Is Loading" src="https://user-images.githubusercontent.com/62723358/84724570-2289fd80-af4e-11ea-875f-1ad22f507250.png">

Extra
* [FP-470 No Data (Jobs & Tickets)](https://user-images.githubusercontent.com/62723358/87967572-a15de300-ca84-11ea-8190-cb4911831b4b.png)
* [FP-470 is Loading (Jobs & Tickets)](https://user-images.githubusercontent.com/62723358/87967914-35c84580-ca85-11ea-9035-5a5083f6cb6d.png)

## Notes: ##
